Documentation/policy.rst

```diff
@@ -0,0 +1,408 @@
+selector.  Either ingress, egress, or both can be provided. If both ingress and
+egress are omitted, the rule has no effect.
+
+::
+
+        type Rule struct {
+                // EndpointSelector selects all endpoints which should be subject to
+                // this rule. Cannot be empty.
+                EndpointSelector EndpointSelector `json:"endpointSelector"`
+
+                // Ingress is a list of IngressRule which are enforced at ingress.
+                // If omitted or empty, this rule does not apply at ingress.
+                //
+                // +optional
+                Ingress []IngressRule `json:"ingress,omitempty"`
+
+                // Egress is a list of EgressRule which are enforced at egress.
+                // If omitted or empty, this rule does not apply at egress.
+                //
+                // +optional
+                Egress []EgressRule `json:"egress,omitempty"`
+
+                // Labels is a list of optional strings which can be used to
+                // re-identify the rule or to store metadata. It is possible to lookup
+                // or delete strings based on labels. Labels are not required to be
+                // unique, multiple rules can have overlapping or identical labels.
+                //
+                // +optional
+                Labels labels.LabelArray `json:"labels,omitempty"`
+
+                // Description is a free form string, it can be used by the creator of
+                // the rule to store human readable explanation of the purpose of this
+                // rule. Rules cannot be identified by comment.
+                //
+                // +optional
+                Description string `json:"description,omitempty"`
+        }
+
+
+*****************
+L3 (Labels Based)
+*****************
+
+The L3 policy specifies which endpoints can talk to each other. L3 policies can
+be specified using Labels_ or CIDR. For CIDR, refer to the policy_cidr_
+section. Label based L3 policy can only be specified on ingress using the
+``fromEndpoints`` field.  ``fromEndpoints`` is a list of endpoints identified
+by a Kubernetes LabelSelector_ which are allowed to communicate with the
+endpoint subject to the rule.
+
+For more complex label combinations, the special ``fromRequires`` can be ued to
+establish label requirements which apply to multiple destinations.
+``fromRequires`` is a list of additional constraints which must be met in order
+for the selected endpoints to be reachable. These additional constraints do no
+by itself grant access privileges and must always be accompanied with at least
+one matching FromEndpoints.
+
+
+Example (Basic)
+===============
+
+::
+
+        [{
+            "endpointSelector": {"matchLabels": {"role":"backend"}},
+            "ingress": [{
+                "fromEndpoints": [
+                  {"matchLabels":{"role":"frontend"}}
+                ]
+            }]
+        }]
+
```

Comment:
I think we should describe what this policy does similar to what we do with `Example (Requires)`